src/Core/src/Platform/Android/MauiSwipeView.cs

```diff
@@ -115,31 +115,36 @@ bool ShouldInterceptTouch(MotionEvent? e)
 
 		bool ShouldInterceptScrollChildrenTouch(SwipeDirection swipeDirection)
 		{
-			if (!(_contentView is ViewGroup viewGroup) || _initialPoint == null)
+			if (_contentView is null || _initialPoint is null)
 				return false;
 
-			int x = (int)(_initialPoint.X * _density);
-			int y = (int)(_initialPoint.Y * _density);
+			var viewGroup = _contentView as ViewGroup;
 
-			bool isHorizontal = swipeDirection == SwipeDirection.Left || swipeDirection == SwipeDirection.Right;
-
-			for (int i = 0; i < viewGroup.ChildCount; i++)
+			if (viewGroup is not null)
 			{
-				var child = viewGroup.GetChildAt(i);
+				int x = (int)(_initialPoint.X * _density);
+				int y = (int)(_initialPoint.Y * _density);
+
+				bool isHorizontal = swipeDirection == SwipeDirection.Left || swipeDirection == SwipeDirection.Right;
 
-				if (child != null && IsViewInBounds(child, x, y))
+				for (int i = 0; i < viewGroup.ChildCount; i++)
 				{
-					if (child is AbsListView absListView)
-						return ShouldInterceptScrollChildrenTouch(absListView, isHorizontal);
+					var child = viewGroup.GetChildAt(i);
 
-					if (child is RecyclerView recyclerView)
-						return ShouldInterceptScrollChildrenTouch(recyclerView, isHorizontal);
+					if (child != null && IsViewInBounds(child, x, y))
+					{
+						if (child is AbsListView absListView)
+							return ShouldInterceptScrollChildrenTouch(absListView, isHorizontal);
 
-					if (child is NestedScrollView scrollView)
-						return ShouldInterceptScrollChildrenTouch(scrollView, isHorizontal);
+						if (child is RecyclerView recyclerView)
+							return ShouldInterceptScrollChildrenTouch(recyclerView, isHorizontal);
 
-					if (child is AWebView webView)
-						return ShouldInterceptScrollChildrenTouch(webView, isHorizontal);
+						if (child is NestedScrollView scrollView)
+							return ShouldInterceptScrollChildrenTouch(scrollView, isHorizontal);
+
+						if (child is AWebView webView)
+							return ShouldInterceptScrollChildrenTouch(webView, isHorizontal);
```

Comment:
Would be a lot nicer with a switch on the type via an expression.

Reply:
Changed.